dask/dataframe/io/sql.py

```diff
@@ -211,3 +213,173 @@ def _read_sql_chunk(q, uri, meta, engine_kwargs=None, **kwargs):
+    dtype : dict or scalar, optional
+        Specifying the datatype for columns. If a dictionary is used, the
+        keys should be the column names and the values should be the
+        SQLAlchemy types or strings for the sqlite3 legacy mode. If a
+        scalar is provided, it will be applied to all columns.
+    compute : bool, default True
+        When true, call dask.compute and perform the load into SQL; otherwise, return a Dask object (or array of
+        per-block objects when parallel=True)
+    parallel : bool, default False
+        When true, have each block append itself to the DB table concurrently. This can result in DB rows being in a
+        different order than the source DataFrame's corresponding rows. When false, load each block into the SQL DB in
+        sequence.
+    kwargs : dict
+        If Pandas version is >= 0.24.0, a "method" value may be passed here; no other kwargs are allowed.
+
+        .. versionadded:: 0.24.0
+
+    Raises
+    ------
+    ValueError
+        When the table already exists and `if_exists` is 'fail' (the
+        default).
+
+    See Also
+    --------
+    read_sql : Read a DataFrame from a table.
+
+    Notes
+    -----
+    Timezone aware datetime columns will be written as
+    ``Timestamp with timezone`` type with SQLAlchemy if supported by the
+    database. Otherwise, the datetimes will be stored as timezone unaware
+    timestamps local to the original timezone.
+
+    .. versionadded:: 0.24.0
+
+    References
+    ----------
+    .. [1] https://docs.sqlalchemy.org
+    .. [2] https://www.python.org/dev/peps/pep-0249/
+
+    Examples
+    --------
+    Create a table from scratch with 4 rows.
+
+    >>> import pandas as pd
+    >>> df = pd.DataFrame([ {'i':i, 's':str(i)*2 } for i in range(4) ])
+    >>> from dask.dataframe import from_pandas
+    >>> ddf = from_pandas(df, npartitions=2)
+    >>> ddf  # doctest: +SKIP
+    Dask DataFrame Structure:
+                       i       s
+    npartitions=2
+    0              int64  object
+    2                ...     ...
+    3                ...     ...
+    Dask Name: from_pandas, 2 tasks
+
+    >>> from dask.utils import tmpfile
+    >>> from sqlalchemy import create_engine
+    >>> with tmpfile() as f:
+    ...     db = 'sqlite:///%s' % f
+    ...     ddf.to_sql('test', db)
+    ...     engine = create_engine(db, echo=False)
+    ...     result = engine.execute("SELECT * FROM test").fetchall()
+    >>> result
+    [(0, 0, '00'), (1, 1, '11'), (2, 2, '22'), (3, 3, '33')]
+    """
+
+    if kwargs:
+        if PANDAS_GT_0240:
+            if list(kwargs.keys()) != ["method"]:
+                raise ValueError(
+                    "Invalid kwargs: %s"
+                    % ",".join([k for k in kwargs.keys() if k != "method"])
+                )
+        else:
+            raise ValueError(
+                "Invalid kwargs: %s" % ",".join([k for k in kwargs.keys()])
+            )
```

Comment:
So ideally this can be replaced with something like

```python
if not PANDAS_GT_0240 and method:
    raise NotImplementedError("'method' requires pandas>=0.24.0. You have version ...")